llvm/include/llvm/Target/GlobalISel/Combine.td

```diff
@@ -1793,7 +1800,7 @@ def all_combines : GICombineGroup<[integer_reassoc_combines, trivial_combines,
     sub_add_reg, select_to_minmax, redundant_binop_in_equality,
     fsub_to_fneg, commute_constant_to_rhs, match_ands, match_ors,
     combine_concat_vector, double_icmp_zero_and_or_combine, match_addos,
-    sext_trunc, zext_trunc, combine_shuffle_concat,
+    sext_trunc, zext_trunc, nneg_zext, combine_shuffle_concat,
```

Comment:
Can we start organizing this into some kind of prefer-signed grouping? Probably should also have a reasonable-but-not-all set that excludes bits like this that are more target dependent